openpgp/packet/private_key_test.go

```diff
@@ -56,6 +65,77 @@ func TestPrivateKeyRead(t *testing.T) {
 	}
 }
 
+func TestPrivateKeyEncrypt(t *testing.T) {
+	var encryptedMsg, decryptedMsg []byte
+	var encryptedC1, encryptedC2 *big.Int
+
+	for i, test := range privateKeyTests {
+		packet, err := Read(readerFromHex(test.privateKeyHex))
+		if err != nil {
+			t.Errorf("#%d: failed to parse: %s", i, err)
+			continue
+		}
+		privKey := packet.(*PrivateKey)
+		err = privKey.Decrypt([]byte("testing"))
+		if err != nil {
+			t.Errorf("#%d: failed to decrypt: %s", i, err)
+			continue
+		}
+
+		switch pubKey := privKey.PublicKey.PublicKey.(type) {
```

Comment:
If I adopt the `ecdh_test.go` approach, I'd have to import `openpgp` and I ended up with an import cycle. So, the compile failed. Any suggestion?